saleor/graphql/product/mutations/products.py

```diff
@@ -40,13 +41,24 @@ def clean_input(cls, info, instance, input, errors):
         cleaned_input = super().clean_input(info, instance, input, errors)
         if 'slug' not in cleaned_input:
             cleaned_input['slug'] = slugify(cleaned_input['name'])
+        parent_id = input['parent']
+        if parent_id:
+            parent = cls.get_node_or_error(
+                info, parent_id, errors, field='parent', only_type=Category)
+            cleaned_input['parent'] = parent
         clean_seo_fields(cleaned_input)
         return cleaned_input
 
     @classmethod
     def user_is_allowed(cls, user, input):
         return user.has_perm('product.manage_products')
 
+    @classmethod
+    def mutate(cls, root, info, **data):
+        parent_id = data.pop('parent_id', None)
+        data['input']['parent'] = parent_id
```

Comment:
I'd opt for sticking to `parent_id` instead of `parent` to keep it consistent

Reply:
I have updated code ~11:40 am CET, but github didn't show this yet. I have just changed to
```
-        data['input']['parent_id'] = parent_id
+        data['input']['parent'] = parent_id
```
and then in `clean_input`
```
-        parent_id = input['parent_id']
+        parent_id = input['parent']
```
Which I think is fine and intuitive.
`clean_input` is supposed to map all data given in `input` in a dictionary, that can be used later (by `construct_instance`) to create an instance of a corresponding model (with proper key names), in particular, mapping GraphQL IDs to an actual objects, for example for a product it would looks something like:
```
_value_given_in_mutation_by_user_            _value_in_cleaned_dict_ 
"Fashionable t-shirt" (string)                "Fashionable t-shirt" (string)
"1hg68sanhcbsgd6a6" (graphql ID)            <Category Clothes> (Category object)
```